jms/1.7/modules/ROOT/pages/jms-studio-configuration.adoc

```diff
@@ -0,0 +1,473 @@
+. Click *OK*.
+. In the *Connection* field select *Generic Connection*.
+. In the *Specification* field select *JMS_2_0*.
+. In the *Connection factory* field select *Edit inline*.
+. Set the *Connection factory jndi name* field to `ConnectionFactory`.
+. In the *Lookup destination* field select any of the following options:
++
+* *NEVER* +
+ No lookup is done and the destinations are created using the existing JMS session.
+* *TRY_ALWAYS* +
+Tries to find the destination using JNDI, and in case it doesn't exist, create the destination using the current JMS session.
+* *ALWAYS* +
+If a queue or topic cannot be found via JNDI, fails with a `JMS:DESTINATION_NOT_FOUND` error.
++
+[start=10]
+. Select the *Name resolver builder* field.
+. Set the *Jdni initial context factory* field to `org.apache.activemq.artemis.ActiveMQInitialContextFactory`.
+. Set the *Jdni provider url* field to `tcp://localhost:61616?broker.persistent=false&amp;broker.useJmx=false`.
+. Set the *Provider properties* to *Edit inline*.
+. Click the plus sign (*+*) to add a new provider property.
+. In the *Provider property* window, set the *Key* field to `queue.jndi-queue-in` and the *Value* field to `in.queue`.
+. Click *Finish*.
+. Repeat step 14.
+. Set the *Key* field to `topic.jndi-topic-in` and the *Value* field to `in.topic`.
+. Click *OK*.
+
+.JMS Connector Generic Configuration
+image::jms-generic-configuration.png[Generic Connection Configuration.]
+
+In the XML editor, the `<jms:generic-connection>` and `<jms:jndi-connection-factory>` configuration looks like this:
+
+[source,xml,linenums]
+----
+<jms:config name="JMS_Config">
+    <jms:generic-connection specification="JMS_2_0">
+        <jms:connection-factory>
+            <jms:jndi-connection-factory connectionFactoryJndiName="ConnectionFactory" lookupDestination="ALWAYS">
+                <jms:name-resolver-builder
+                        jndiInitialContextFactory="org.apache.activemq.artemis.ActiveMQInitialContextFactory"
+                        jndiProviderUrl="tcp://localhost:61616?broker.persistent=false&amp;broker.useJmx=false">
+                    <jms:provider-properties>
+                        <jms:provider-property key="queue.jndi-queue-in" value="in.queue"/>
+                        <jms:provider-property key="topic.jndi-topic-in" value="in.topic"/>
+                    </jms:provider-properties>
+                </jms:name-resolver-builder>
+            </jms:jndi-connection-factory>
+        </jms:connection-factory>
+    </jms:generic-connection>
+</jms:config>
+----
+
+=== Configure the Connection Libraries
+
+When you are configuring any connection type, you must always configure a library containing the JMS client implementation since JMS Connector is not bound to any particular implementation. There are ActiveMQ external libraries and generic external libraries that you can configure in the global elements view of Studio or manually adding the library dependency in your `pom.xml` file.
```

Comment:
```suggestion
When you are configuring any connection type, you must always configure a library containing the JMS client implementation since JMS Connector is not bound to any particular implementation. There are ActiveMQ external libraries and generic external libraries that you can configure in the global elements view of Studio or by manually adding the library dependency in your `pom.xml` file.
```